rust/core-lib/src/index_set.rs

```diff
@@ -85,6 +124,30 @@ impl IndexSet {
         }
     }
 
+    /// Computes a new set based on applying a delta to the old set.
```

Comment:
Maybe a bit more detail of how the delta is applied, e.g. that collapsed regions are removed and contiguous regions are combined?